device/cuda/include/traccc/cuda/seeding/seed_finding.hpp

```diff
@@ -24,112 +24,129 @@
 
 // System include(s).
 #include <algorithm>
-#include <iostream>
-#include <mutex>
 
 namespace traccc {
 namespace cuda {
 
 /// Seed finding for cuda
 struct seed_finding : public algorithm<host_seed_collection(
-                          host_spacepoint_container&&, sp_grid&&)> {
+                          host_spacepoint_container&&, sp_grid_buffer&&)> {
```

Comment:
This algorithm should take a view object (`sp_grid_view`) like the other functions do. Functions/classes should basically never receive data or buffer objects. They should always receive view objects. The caller can then decide whether they want to give the function/class a view on top of a host object, a view on top of a buffer, or whatever else...

Reply:
Resolved